### ISSUE TITLE
How to Calculated Highest-Recommended item(s) across groups of users/rows?

### ISSUE THREAD

**Issue by MaxPowerWasTaken** [lyst/lightfm#640]
I have a set of LightFM recommendations I have trained via WARP loss and just implicit feedback (user-item transaction history; no separate user-features or item-features yet). 

I have groups of users/rows that are relevant for my business case, and I want to calculate the "most-recommended (or top 3) items across [group of users]." However, I've seen warnings in other comments that recommendation scores seem not to be comparable across user/rows (e.g. [maciejkula](https://github.com/maciejkula) wrote "User-item scores out of the LightFM model do not have a guaranteed range or scale (as that is irrelevant for ranking items for any particular user)" in 2018 in [this issue](https://github.com/lyst/lightfm/issues/242) 

Is there a recommended way to calculate e.g. top-1,2,3 recommended items across groups of users, since we shouldn't average user-item recommendation scores across user/rows? A couple possible ways forward below, but wanted to ask the experts in case I'm off-base somehow:
- Normalizing user-item recommendation scores within rows (so each user/row's item scores vary -1 to 1), and then averaging those normalized values across rows (maciejkula recommended something like this in the issue above, albeit in response to a different usecase/question)
- Just counting the number of users an item is top-ranked for. Top-ranked item across users is item that gets most highest-recommendation-score. (Though this would not provide credit/weighting for items which are consistently e.g. 2nd-highest-recommended instead of 20th-highest-recommended item for many users).

Any advice/guidance or corrections to anything I may have misunderstood would be much appreciated. Thanks for maintaining this great library!

**Comment by Ganners** [comment]
I'm not sure how you wish to partition the groups, perhaps you know the indices of users which correspond to that split already.

If you're looking to just find the top recommended items then you should just need to argsort, pick some cutoff (I.e. how many you recommendations you display to the user), and count the ids.

```python
from lightfm import LightFM
from lightfm.datasets import fetch_movielens
import numpy as np

n_displayed_to_user = 10

movielens = fetch_movielens()
train = movielens['train']
test = movielens['test']

model = LightFM(learning_rate=0.05, loss='warp')
model.fit_partial(train, epochs=10)

# or use the predict function and stack up a matrix of scores
scores = (model.user_embeddings @ model.item_embeddings.T + model.item_biases).T + model.user_biases
topk = (-scores).argsort(axis=1)[:, :n_displayed_to_user]
ids, counts = np.unique(topk.flatten(), return_counts=True)
df = pd.DataFrame({'item_id': ids, 'count': counts}).sort_values('count', ascending=False)
df
```
